contracts/RelayedDocumentStore.sol

```diff
@@ -0,0 +1,112 @@
+  *     The trusteSigner is the issuing party that is required to sign the actual messages and transaction data.
+  *     This contract then relays transactions to the OpenAttestation DocumentStore on behalf of the trusedSigner.
+  *     The most notable parameters that have been added to extend the DocumentStore are:
+  *       1. nonce: Leveraged in order to protect against replay attacks although this risk is quite low as issuance / revocation
+  *                 may only occur once by design.
+  *       2. documentsHash: The hash of an array of documentMerkleRoots that are being bulk issued / revoked. Only for bulk operations.
+  *                         - const message = web3.utils.soliditySha3(...documentMerkleRoots);
+  *       3. signature: The off-chain signature, signed with the privateKey from the trustedSigner EOA, that will be used to
+  *                     authenticate the trustedSigner and verify the transaction.
+  *     ---
+  *     The process for generating a valid signature is as follows:
+  *       1. Generate the message to be signed. This will be either the docuementMerkleRoot or the documentsHash of an array of docuementMerkleRoots (for batch operations).
+  *         - const message = "0x3a267813bea8120f55a7b9ca814c34dd89f237502544d7c75dfd709a659f6330"; // documentMerkleRoot
+  *         OR
+  *         - const documentMerkleRoots = ["0x3a267813bea8120f55a7b9ca814c34dd89f237502544d7c75dfd709a659f6330", "...", ...];
+  *           const message = web3.utils.soliditySha3(...documentMerkleRoots); // hash of many documentMerkleRoots for a batch operation
+  *       2. Generate the nonce... transaction count... random number etc.
+  *       3. Generate the signature.  This signature is generated by signing the hash of: the nonce, the message as well as the relay contract addres.
+  *          The relay contract address is used to also elimiate the risk of replay attacks across other relay contracts.
+  *          The resulting signature should be in hex format following the Ethereum signature encoding.
+  *          For example:
+  *           - const message = "0x3a267813bea8120f55a7b9ca814c34dd89f237502544d7c75dfd709a659f6330";
+  *             const nonce = 1;
+  *             const address = contract.address; // 0x8CdaF0CD259887258Bc13a92C0a6dA92698644C0
+  *             const messageHash = web3.utils.soliditySha3(message, nonce, address); // 0x2bd7dce779d8d21e3266e847f21ce8160a53e6a78a65a5045f72b18081f7fc69
+  *             const signature = await web3.eth.sign(messageHash, signer); // 0xf5731da089e7d9532aad446f9f92cabf3ce39c5cde8ba541cb26ed17b69ab8870a411a66ad8316d7b7478e1c1be05ca840ecfc8cc7468beb9c2832be4dc8fb1e1b
+ */
+contract RelayedDocumentStore is DocumentStore {
+  using ECDSA for bytes32;
+
+  address private trustedSigner; // Account that is signing *messages*
+
+  function initialize(
+    string memory _name,
+    address _trustedSigner,
+    address _relayer
```

Comment:
_relayer doesn't seem to be checked anywhere - is it intended that anyone should be able to relay?

Reply:
I think anyone should be able to be defined as a relaying address yes. Checking just against 0 at this point.